### PR TITLE
Update all variables so long as they're in the mapped document

### PR DESCRIPTION
**Pull request by distantcam**
OctopusDeploy/Issues#6029

https://octopusdeploy.slack.com/archives/CNHBHV2BX/p1574314747210800

As per slack discussion, removing the prefix filter.

### CODE REVIEW

**Comment by distantcam** [comment]
[Allow variable replacement when variables are prefixed by Octopus](https://trello.com/c/KnDtfwmG/2555-allow-variable-replacement-when-variables-are-prefixed-by-octopus)

**Comment by distantcam** [comment]
@droyad I've added to the checks to also look for a colon `:` in the variable after octopus, so that existing scenarios won't be broken. I've also added some test cases around this. Does this address your concern?

**Comment by droyad** [comment]
/cc @MJRichardson 

I think we should document this behaviour and maybe add a log message if there is a root property in the json called "Octopus".

File: source/Calamari.Shared/Integration/JsonVariables/JsonConfigurationVariableReplacer.cs [hunk 1]
```diff
@@ -182,18 +182,26 @@ private void PopPath()
...
-                    {
-                        map[name](variables.Get(name));
-                    }
-                    catch (Exception e)
-                    {
-                        Log.WarnFormat("Unable to set value for {0}. The following error occurred: {1}", name, e.Message);
-                    }
+                    // Only include variables starting with 'Octopus'
+                    // if it also has a colon (:)
+                    return v.StartsWith("Octopus:");
```
Comment (slewis74) [inline source/Calamari.Shared/Integration/JsonVariables/JsonConfigurationVariableReplacer.cs]: Should there be a `StringComparison.OrdinalIgnoreCase` here?

File: source/Calamari.Shared/Integration/JsonVariables/JsonConfigurationVariableReplacer.cs [hunk 2]
```diff
@@ -182,18 +182,26 @@ private void PopPath()
...
+                }
+                return map.ContainsKey(v);
+            }
```
Comment (slewis74) [inline source/Calamari.Shared/Integration/JsonVariables/JsonConfigurationVariableReplacer.cs]: Can we rename `filter` to `VariableNameIsNotASystemVariable`?

File: source/Calamari.Shared/Integration/JsonVariables/JsonConfigurationVariableReplacer.cs [hunk 3]
```diff
@@ -182,18 +182,26 @@ private void PopPath()
...
-                    }
-                    catch (Exception e)
-                    {
-                        Log.WarnFormat("Unable to set value for {0}. The following error occurred: {1}", name, e.Message);
-                    }
+                    // Only include variables starting with 'Octopus'
+                    // if it also has a colon (:)
+                    return v.StartsWith("Octopus:");
+                }
+                return map.ContainsKey(v);
```
Comment (slewis74) [inline source/Calamari.Shared/Integration/JsonVariables/JsonConfigurationVariableReplacer.cs]: I think there is a logic gap here in the filter method. In the case where it starts with `Octopus:` should we not then be doing a `map.ContainsKey(v)`? I was expecting to just see something like ` || x.StartsWith("Octopus:", StringComparisn.OrdinalIgnoredCase` added to the original Where clause

**Pull request merged**